infra/enrolment-records-store/src/test/java/com/simprints/infra/enrolment/records/store/commcare/CommCareIdentityDataSourceTest.kt

```diff
@@ -146,25 +172,28 @@ class CommCareIdentityDataSourceTest {
         every { mockDataCursor.moveToNext() } returns true
         every { mockDataCursor.getColumnIndexOrThrow(COLUMN_DATUM_ID) } returns 0
         every { mockDataCursor.getColumnIndexOrThrow(COLUMN_VALUE) } returns 1
-        every { mockDataCursor.getString(0) } returnsMany
-                listOf("someOtherDatumId", "subjectActions", "someOtherDatumId", "subjectActions")
-        every { mockDataCursor.getString(1) } returnsMany
-                listOf(SUBJECT_ACTIONS_FINGERPRINT_1, SUBJECT_ACTIONS_FINGERPRINT_2)
+        every { mockDataCursor.getString(0) } returnsMany listOf(
+            "someOtherDatumId", "subjectActions", "someOtherDatumId", "subjectActions"
+        )
+        every { mockDataCursor.getString(1) } returnsMany listOf(
+            SUBJECT_ACTIONS_FINGERPRINT_1, SUBJECT_ACTIONS_FINGERPRINT_2
+        )
 
-        val query = SubjectQuery()
+        val templateFormat = "ISO_19794_2"
+        val query = SubjectQuery(fingerprintSampleFormat = templateFormat)
         val range = 0..expectedFingerprintIdentities.size
         val actualIdentities = dataSource.loadFingerprintIdentities(query, range)
 
-        assertEquals(expectedFingerprintIdentities.size, actualIdentities.size)
-        val areContentsEqual = expectedFingerprintIdentities.zip(actualIdentities) { expected, actual ->
-            expected.subjectId == actual.subjectId &&
-            expected.fingerprints.zip(actual.fingerprints) { expectedFingerprint, actualFingerprint ->
-                expectedFingerprint.fingerIdentifier == actualFingerprint.fingerIdentifier &&
-                expectedFingerprint.templateQualityScore == actualFingerprint.templateQualityScore &&
-                expectedFingerprint.template.contentEquals(actualFingerprint.template) &&
-                expectedFingerprint.format == actualFingerprint.format
-            } .all { it }
-        } .all { it }
+        assertEquals(1, actualIdentities.size)
+        val areContentsEqual =
+            expectedFingerprintIdentities.filter { identity -> identity.fingerprints.any { it.format == templateFormat } }
+                .zip(actualIdentities) { expected, actual ->
```

Comment:
Minor formatting nitpick: this block is extremely confusing with its wrapping and mixing of comparisons and function chains.

Reply:
Maybe you could make use of https://truth.dev/api/1.1/index.html?com/google/common/truth/Correspondence.html to simplify the checks here?